packages/qgrid-ngx/src/lib/row/tr-core.directive.ts

```diff
@@ -15,7 +16,7 @@ import { GridPlugin } from '../plugin/grid-plugin';
 export class TrCoreDirective implements DomTr, OnInit, OnDestroy {
   @Input('q-grid-core-index') viewIndex: number;
   @Input('q-grid-core-tr') model: any;
-  @Input('q-grid-core-source') source;
+  @Input('q-grid-core-source') source: string;
```

Comment:
```typescript
type BoxParts = keyof { body: Bag; head: Bag; foot: Bag };
OR
type BoxParts = 'body' | 'head' | 'foot';
...
@Input('q-grid-core-source') source: BoxParts;
...
this.plugin.table.box.bag[this.source as BoxParts]
```
Btw, check if there's already an interface that describe this thing `{ body: Bag; head: Bag; foot: Bag }` and use it if exist